src/main.c

```diff
@@ -2123,9 +2123,7 @@ static gboolean service_start(int argc, char **argv)
 			if (g_strcmp0(r_context()->bootslot, "_external_") == 0) {
 				r_event_log_booted_external();
 			} else {
-				RaucSlot *booted_slot = r_slot_find_by_device(r_context()->config->slots, r_context()->bootslot);
-				if (!booted_slot)
-					booted_slot = r_slot_find_by_bootname(r_context()->config->slots, r_context()->bootslot);
+				RaucSlot *booted_slot = r_slot_get_booted(r_context()->config->slots);
```

Comment:
Do we still need the "_external_" workaround from 2928ac0603a77c7e817d541f6a872fc81ad59258 then?

Reply:
Yes, since for *external*, there is no real 'booted' slot that we can get.

When I originally implemented the surrounding code, I used an approach that generated a dummy slot for the 'external' case. With this, the special handling for *external* would have been obsolete. But this was dropped during review since it might have had other impact.

Reply:
ok, thanks